Sources/Basics/Sandbox.swift

```diff
@@ -73,8 +74,13 @@ fileprivate func macOSSandboxProfile(
     }
     // Optionally allow writing to temporary directories (a lot of use of Foundation requires this).
     else if strictness == .writableTemporaryDirectory {
-        writableDirectoriesExpression.append("(subpath \"/private/tmp\")")
-        if let tmpDir = try? TSCBasic.determineTempDirectory() {
+        // Add the standard and Foundation temporary directories, and the one determined by TSC (which also taked into account environment variables).
+        var temporaryDirectories = Set([AbsolutePath("/tmp"), AbsolutePath(NSTemporaryDirectory())])
```

Comment:
Note that this will get resolved for `/private/tmp` since all the paths now go through the `resolveSymlinks` below.